supertokens_python/recipe/session/access_token.py

```diff
@@ -41,24 +45,62 @@ def sanitize_number(n: Any) -> Union[Union[int, float], None]:
 
 
 def get_info_from_access_token(
-    jwt_info: ParsedJWTInfo, jwt_signing_public_key: str, do_anti_csrf_check: bool
+    jwt_info: ParsedJWTInfo,
+    jwk_clients: List[JWKClient],
+    do_anti_csrf_check: bool,
 ):
+    # TODO: Add different tests to verify this works as expected
     try:
-        verify_jwt(jwt_info, jwt_signing_public_key)
-        payload = jwt_info.payload
-
-        validate_access_token_structure(payload)
+        payload: Optional[Dict[str, Any]] = None
+
+        if jwt_info.version < 3:
+            # It won't have kid. So we'll have to try the token against all the keys from all the jwk_clients
+            for client in jwk_clients:
```

Comment:
We should only use the first available client; they are all expected to return the same value. Combining them is to have fallbacks in case a core instance goes down.